server/src/main/java/org/elasticsearch/index/search/QueryStringQueryParser.java

```diff
@@ -681,7 +681,9 @@ private Query getWildcardQuerySingle(String field, String termStr) throws ParseE
                 setAnalyzer(forceAnalyzer);
                 return super.getWildcardQuery(currentFieldType.name(), termStr);
             }
-
+            if (getAllowLeadingWildcard() == false && (termStr.startsWith("*") || termStr.startsWith("?"))) {
```

Comment:
We should add the same check for prefix query too ?

Reply:
I'm confused how leading wildcard characters can start a prefix query. Wouldn't that only be "*" and "?", and wouldn't those get resolved as wildcard expressions?

Reply:
I was confused because the QueryParserBase does the check on prefix query too (`field:*`) but we intercept these queries in ES to transform them into an exists query so there's no issue here. Sorry for the confusion.